### PR TITLE
chapter3

### PR DESCRIPTION
**Pull request by xtrxllx**
ready

### CODE REVIEW

**Comment by xtrxllx** [comment]
ready

File: app/assets/stylesheets/chapter3.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the chapter3 controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/chapter3.scss]: remove file

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -12,4 +12,7 @@
  *
  *= require_tree .
  *= require_self
- */
+*/
+
+@import "bootstrap-sprockets";
+@import "bootstrap";
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.scss]: using '

File: app/controllers/chapter3_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Chapter3Controller < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (saobv-2223) [inline app/controllers/chapter3_controller.rb]:
```suggestion
  def home; end

  def help; end
```

File: app/controllers/chapter3_mostly_static_pages_controller.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Chapter3MostlyStaticPagesController < ApplicationController
+  def home
+  end
+
+  def help
+  end
```
Comment (saobv-2223) [inline app/controllers/chapter3_mostly_static_pages_controller.rb]: same above

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+    end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]: indenting 2 spaces

**Comment by xtrxllx** [comment]
ready

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
+  end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
    base_title = "Ruby on Rails Tutorial Sample App"
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
  end
```

File: app/assets/stylesheets/custom.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    padding-top: 60px;
+    }
+    section {
+    overflow: auto;
+    }
+    textarea {
+    resize: vertical;
+    } 
+    center {
+    text-align: center;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: image

File: app/helpers/chapter3_helper.rb [hunk 8]
```diff
@@ -0,0 +1,2 @@
+module Chapter3Helper
+end
```
Comment (saobv-2223) [inline app/helpers/chapter3_helper.rb]: remove file that never used

File: app/views/chapter3/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<% provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a href="http://www.railstutorial.org/help">Rails Tutorial
+help page</a>.
```
Comment (saobv-2223) [inline app/views/chapter3/help.html.erb]: using link_to instead of a tag

File: app/views/chapter3/home.html.erb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<% provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<a href="http://www.railstutorial.org/">Ruby on
+Rails Tutorial</a>
```
Comment (saobv-2223) [inline app/views/chapter3/home.html.erb]: same above

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -1,15 +1,18 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>SampleApp</title>
+    <title><%= full_title(yield(:title)) %></title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]:
```suggestion
    <title><%= full_title yield(:title) %></title>
```

**Comment by xtrxllx** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
Your PR is having 2 commits, merge to 1 commit please

**Comment by xtrxllx** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -0,0 +1,40 @@
+/* universal */
+
+body {
+    padding-top: 60px;
+    }
+
+    section {
+    overflow: auto;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: indenting only 2 spaces

File: app/views/chapter3/help.html.erb [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<% provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<a link_to "http://www.railstutorial.org/help">Rails Tutorial
```
Comment (saobv-2223) [inline app/views/chapter3/help.html.erb]: using link_to instead of a tag

File: app/views/chapter3/home.html.erb [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<% provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<a link_to "http://www.railstutorial.org/">Ruby on
```
Comment (saobv-2223) [inline app/views/chapter3/home.html.erb]: using link_to instead of a tag

**Comment by xtrxllx** [comment]
ready

File: app/views/chapter3/help.html.erb [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<% provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+Get help on the Ruby on Rails Tutorial at the
+<%= link_to "Rails Tutorial help page" , "http://www.railstutorial.org/help" %>.
+
+To get help on this sample app, see the
+<%= link_to "Ruby on Rails Tutorial" , "http://www.railstutorial.org/book"%>
+book.
```
Comment (saobv-2223) [inline app/views/chapter3/help.html.erb]: indenting 2 spaces

File: app/views/chapter3/home.html.erb [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+This is the home page for the
+<%= link_to "Ruby on Rails Tutorial" , "http://www.railstutorial.org/" %>
+sample application.
```
Comment (saobv-2223) [inline app/views/chapter3/home.html.erb]: indenting 2 spaces

File: config/initializers/config.rb [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+  
+    # Validate presence and type of specific config values. Check https://github.com/dry-rb/dry-validation for details.
+    #
+    # config.schema do
+    #   required(:name).filled
+    #   required(:age).maybe(:int?)
+    #   required(:email).filled(format?: EMAIL_REGEX)
+    # end
+  
+  end
```
Comment (saobv-2223) [inline config/initializers/config.rb]: add an empty line below line 55

**Comment by xtrxllx** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**